### DISCUSSION TITLE
Path instance returns own path as '/c:/Users/', even if it's normalized, on windows

### DISCUSSION THREAD

**Discussion by cenkce** [eclipse-theia/theia#9675] [General]
the path causes errors, what do you recommend?

**Comment by msujew** [comment]
Hi @cenkce, would you mind sharing the code how you obtain your path?

In general, you can call the [`FileUri.fsPath`](https://github.com/eclipse-theia/theia/blob/87b5b526d4aea461a54cc5f9111184b86e9be422/packages/core/src/node/file-uri.ts#L38-L59) function to retrieve the file system specific file path from an URI.

**Reply by cenkce** [reply]
There is a property that name is codeUri inside of an URI instance but the property is not exposed in the URI class definition. How do you use it to get valid fsPath from an URI?

```
codeUri: _URI
  authority: ""
  fragment: ""
  fsPath: "c:\Users\user\Documents\ws"
  path: "/c:/Users/user/Documents/ws"
  query: ""
  scheme: "file"
```

**Comment by cenkce** [comment]
```
    const wsFileStat: FileStat | undefined = this.workspaceService.workspace; // WorkspaceService
    if (wsFileStat) {
      this._workspacePath = wsFileStat.resource.path; // this path 
```

**Reply by msujew** [reply]
Thanks for providing some code context! You can simply call the static [`FileUri.fsPath`](https://github.com/eclipse-theia/theia/blob/87b5b526d4aea461a54cc5f9111184b86e9be422/packages/core/src/node/file-uri.ts#L38-L59) function to retrieve the file system specific file path from an URI. You can use `wsFileStat.resource` as the argument for that.

**Reply by cenkce** [reply]
Ok thanks
